### PR TITLE
Updated AWS add extension nodes

### PR DESCRIPTION
**Pull request by ianf77**
Added files (see PR for notes)

### CODE REVIEW

**Review by zkayyali812 — APPROVED** [review]

File: stories/topics/proc-aap-aws-deploying-extension-nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+.Procedure
+. To deploy the extension nodes, run the command generator to generate the CLI command.
++
+[source,bash]
+---- 
+$ docker run --rm -v $(pwd)/command_generator_data:/data $IMAGE command_generator --data-file /data/extra_vars.yml
+----
++
+Provides the following command:
```
Comment (ariordan-redhat) [inline stories/topics/proc-aap-aws-deploying-extension-nodes.adoc:11]:
```suggestion
Provides the following upgrade command:
```
Just a suggestion: the next step in the procedure refers to this as the upgrade command

File: stories/assembly-aap-aws-deploying-extension-nodes.adoc [hunk 2]
```diff
@@ -5,8 +5,33 @@ ifdef::context[:parent-context: {context}]
...
+
 include::topics/ref-aap-aws-decide-offer-type.adoc[leveloffset=+1]
-include::topics/proc-aap-aws-prepare-environment.adoc[leveloffset=+1]
-include::topics/proc-aap-aws-create-extension-nodes.adoc[leveloffset=+1]
+//The next two files  are duplicates of those in PR#86. Merge PR#86 first and then these files will be available.
+//Remove comments and comment markers.
+//include::topics/con-aap-pull-container-image.adoc[leveloffset=+1]
+//include::topics/ref-aap-aws-permissions-to-manage-nodes.adoc[leveloffset=+1]
+include::topics/proc-aap-aws-generate-add-data-files.adoc[leveloffset=+1]
+include::topics/proc-aa-aws-update-add_data-files.adoc[leveloffset=+1]
```
Comment (ariordan-redhat) [inline stories/assembly-aap-aws-deploying-extension-nodes.adoc:32]:
```suggestion
include::topics/proc-aa-aws-update-add-data-files.adoc[leveloffset=+1]
```

File: stories/assembly-aap-aws-deploying-extension-nodes.adoc [hunk 3]
```diff
@@ -5,8 +5,33 @@ ifdef::context[:parent-context: {context}]
...
+//Don't need these
+//include::topics/proc-aap-aws-prepare-environment.adoc[leveloffset=+1]
+//include::topics/proc-aap-aws-create-extension-nodes.adoc[leveloffset=+1]
+include::topics/proc-aap-aws-deploying_extension-nodes.adoc[leveloffset=+1]
```
Comment (ariordan-redhat) [inline stories/assembly-aap-aws-deploying-extension-nodes.adoc:36]:
```suggestion
include::topics/proc-aap-aws-deploying-extension-nodes.adoc[leveloffset=+1]
```

**Review by ariordan-redhat — CHANGES_REQUESTED** [review]

File: stories/topics/proc-aap-aws-deploying-extension-nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+. Run the supplied upgrade command to add the extension nodes.
++
+[source,bash]
+----
+$ docker run --rm --env PLATFORM=AWS -v ~/.aws/credentials:/home/runner/.aws/credentials \
+--env ANSIBLE_CONFIG=../aws-ansible.cfg --env DEPLOYMENT_NAME=AnsibleAutomationPlatform \
+--env GENERATE_INVENTORY=true  $IMAGE redhat.ansible_on_clouds.aws_upgrade \
+-e 'aws_foundation_stack_name=AnsibleAutomationPlatform aws_region=us-east-1 \
+aws_s3_bucket=AnsibleAutomationPlatform-bucket aws_ssm_bucket_name=AnsibleAutomationPlatform-bucket \ aws_backup_vault_name=Default \
+aws_backup_iam_role_arn=arn:aws:iam::<YOUR_ACCOUNT_NUMBER>:role/service-role/AWSBackupDefaultServiceRole'
```
Comment (zkayyali812) [inline stories/topics/proc-aap-aws-deploying-extension-nodes.adoc:33]:
```suggestion
docker run --rm --env PLATFORM=AWS -v ~/.aws/credentials:/home/runner/.aws/credentials \
--env ANSIBLE_CONFIG=../aws-ansible.cfg  $IMAGE redhat.ansible_on_clouds.aws_add_extension_nodes \
-e 'aws_foundation_stack_name=AnsibleAutomationPlatform aws_region=us-east-1 \ aws_launch_template_name=AnsibleAutomationPlatform-ext-lt1-100 \ 
aws_autoscaling_group_name=AnsibleAutomationPlatform-ext-asg1-100 \
aws_asg_min_size=2 aws_asg_desired_capacity=2 aws_offer_type=100'
```

File: stories/topics/proc-aap-aws-deploying-extension-nodes.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+.Procedure
+. To deploy the extension nodes, run the command generator to generate the CLI command.
++
+[source,bash]
+---- 
+$ docker run --rm -v $(pwd)/command_generator_data:/data $IMAGE command_generator --data-file /data/extra_vars.yml
+----
++
+Provides the following upgrade command:
```
Comment (zkayyali812) [inline stories/topics/proc-aap-aws-deploying-extension-nodes.adoc:11]: Please remove the word `upgrade` from this line

**Review by zkayyali812 — CHANGES_REQUESTED** [review]
Not an upgrade procedure, remove word upgrade where used

File: stories/topics/proc-aap-aws-generate-add-data-files.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
++
+[source,bash]
+----
+$ docker run --rm -v $(pwd)/command_generator_data:/data $IMAGE \
+command_generator_vars aws_add_extension_nodes \
+--output-data-file /data/extra_vars.yml
+----
+
+. When you have run these commands, a `command_generator_data/extra_vars.yml` template file is created. 
+This template file resembles the following:-
```
Comment (ariordan-redhat) [inline stories/topics/proc-aap-aws-generate-add-data-files.adoc:25]:
```suggestion
This template file resembles the following:
```

File: stories/topics/proc-aap-aws-update-add-data-files.adoc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+= Update the data file
+
+You must populate the data file before triggering the operation. 
+The variables listed in the data file are defined below.
+
+* `cloud_credentials_path` is the path to your AWS credentials file.
+For example, `~/.aws/credentials`
+* `deployment_name` is the name of the foundation deployment. 
+This is the same name you used when you deployed the foundation.
+* `aws_asg_desired_capacity` is the AutoScalin Group desired capacity.
```
Comment (ariordan-redhat) [inline stories/topics/proc-aap-aws-update-add-data-files.adoc:12]:
```suggestion
* `aws_asg_desired_capacity` is the AutoScaling Group desired capacity.
```

File: stories/topics/proc-aap-aws-update-add-data-files.adoc [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+* `aws_asg_min_size` is the AutoScaling group minimum capacity. The maximum is set by `aws_offer-type`.
+* `aws_autoscaling_group_name` is the name of the AWS AutoScaling Group to create for the extension nodes.
+* `aws_launch_template_name` is the name of the AWS EC2 launch template to create.
+* `aws_offer_type` is the offer type of the extension nodes. This must be `100`, `200`, or `400`.
+* `aws_region` is the region where the foundation deployment is located.
+
+After populating the data file, it resembles the following:
+
+The values are provided as examples:
```
Comment (ariordan-redhat) [inline stories/topics/proc-aap-aws-update-add-data-files.adoc:21]:
```suggestion
The values are provided as examples.
```

File: stories/topics/con-aap-aws-pull-deploy-container-image.adoc
Comment (ariordan-redhat) [inline stories/topics/con-aap-aws-pull-deploy-container-image.adoc:1]: Can we delete this file and remove the `include` from `assembly-aap-aws-deploying-extension-nodes.adoc`, or are you planning to add content to the module later ?

**Review by ariordan-redhat — COMMENTED** [review]
A few suggestions

**Review by ariordan-redhat — COMMENTED** [review]
A few suggestions

File: stories/topics/con-aap-aws-pull-deploy-container-image.adoc
Comment (ianf77) [reply stories/topics/con-aap-aws-pull-deploy-container-image.adoc:1]: Sorry. I added the content.

**Pull request merged**